### PR TITLE
Storage - Make UriBuilders public

### LINKED ISSUE #6885: Storage: Make UriBuilders public
The various `*UriBuilder` types for constructing URIs that move up and down the Storage service hierarchy are very useful.  We should expose them publicly for users who might need them as well.

### PR DESCRIPTION
**Pull request by tg-msft**
Fixes #6885.  As part of this, it changes the private UriBuilders to look more like UriBuilder and RequestUriBuilder.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs [hunk 1]
```diff
@@ -4,99 +4,163 @@
...
+        }
+        private string _snapshot;
 
         ///// <summary>
-        ///// VersionId.  Empty string if not present in URI.
+        ///// Gets or sets the VersionId.  The value defaults to
+        ///// <see cref="String.Empty"/> if not present in the <see cref="Uri"/>.
         ///// </summary>
-        //public string VersionId;
+        //public string VersionId
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs]: Why is this one commented out?

File: sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs [hunk 2]
```diff
@@ -163,142 +227,99 @@ public BlobUriBuilder(Uri uri)
...
-            //// Concatenate blob version query parameter (if it exists)
             //if (!String.IsNullOrWhiteSpace(this.VersionId))
             //{
-            //    if (rawQuery.Length > 0)
-            //    {
-            //        rawQuery += "&";
-            //    }
-
-            //    rawQuery += VersionIdParameterName + "=" + this.VersionId;
+            //    if (query.Length > 0) { query += "&"; }
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs]: Why is this commented out?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs]: It's a feature we don't fully support yet.  It was commented out in the private UriBuilders so I updated the comments to make it easier for folks to add back when they're ready.

File: sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs (hunk 2 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs]: Same as above

**Review by seanmcc-msft — APPROVED** [review]

**Comment by tg-msft** [comment]
Filed #7539 to derive these from RequestUriBuilder in the near future.

**Pull request merged**